src/WebApiContrib.Core.Versioning/RouteDataVersionStrategy.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+
+namespace WebApiContrib.Core.Versioning
+{
+    /// <summary>
+    /// This versioning strategy gets its version from the resource URI.
+    /// It picks up the version from a captured route value, called <c>version</c>.
+    /// </summary>
+    public class RouteDataVersionStrategy : IVersioningStrategy
+    {
+        /// <inheritdoc />
+        public int? GetVersion(HttpContext context, RouteData routeData)
+        {
+            if (routeData == null)
+            {
+                throw new ArgumentNullException(nameof(routeData));
+            }
+
+            var routeValues = routeData.Values;
+
+            object versionObject;
+            if (!routeValues.TryGetValue("version", out versionObject))
```

Comment:
Yeah, I'll see if I can find a way to configure it. I don't want to hang it directly off the options object because it's specific to only this version strategy.